### ISSUE TITLE
Metric definitions for message/event networking

### ISSUE THREAD

**Issue by kmcclellan** [open-telemetry/semantic-conventions#120]
I noticed there are [semantic definitions for Kafka metrics](https://github.com/open-telemetry/semantic-conventions/blob/main/specification/metrics/semantic_conventions/instrumentation/kafka.md), many of which could apply to other technologies that send messages/events over a network (AMQP, for example).

It seems valuable to define these metrics in way that is agnostic to technology/protocol, similar to the [RPC metrics](https://github.com/open-telemetry/semantic-conventions/blob/main/specification/metrics/semantic_conventions/rpc-metrics.md) (in which there exists a similar overlap with HTTP).

**Comment by joaopgrassi** [comment]
Hi @kmcclellan the messaging working group is tasked to work on "general" messaging metrics as part of the stabilization work. @lmolkova opened a draft PR with some ideas https://github.com/open-telemetry/semantic-conventions/pull/163. Sorry this slipped and wasn't replied earlier. Let us know if you have other ideas and feel free to leave comments on the PR or join our SIG meetings.

**Comment by pyohannes** [comment]
For a first stable version of messaging semantic conventions we will focus on client-side instrumentation (producer and consumer) of traces and metrics.

Broker instrumentation is something that will be tackled post-v1.

**Comment by kmcclellan** [comment]
Makes sense, esp. given not all messaging protocols involve a "broker" role. Generally it might make sense to avoid Kafka-specific language for these semantics. "Publisher" and "subscriber" might serve better to that end (as opposed to "producer"/"consumer"). "Message" probably serves better than "event."

"Client" too may be worthy of avoidance, since pub/sub is quite a different paradigm than client/server (data flows in one direction instead of a bidirectional request/response).

On second thought, I better just take a look at the PR 🤠
